py/torch_tensorrt/dynamo/conversion/_conversion.py

```diff
@@ -3,34 +3,26 @@
 import io
 from typing import Sequence
 
+import tensorrt as trt
 import torch
 from torch_tensorrt._Input import Input
 from torch_tensorrt.dynamo._settings import CompilationSettings
-from torch_tensorrt.dynamo.conversion._TRTInterpreter import TRTInterpreter
+from torch_tensorrt.dynamo.conversion._TRTInterpreter import (
+    TRTInterpreter,
+    TRTInterpreterResult,
+)
 from torch_tensorrt.dynamo.runtime import PythonTorchTensorRTModule, TorchTensorRTModule
-from torch_tensorrt.dynamo.utils import get_torch_inputs
-
-import tensorrt as trt
+from torch_tensorrt.dynamo.utils import get_torch_inputs, to_torch_device
 
 
-def convert_module(
+def interpret_module(
     module: torch.fx.GraphModule,
     inputs: Sequence[Input],
     settings: CompilationSettings = CompilationSettings(),
     name: str = "",
-) -> PythonTorchTensorRTModule | TorchTensorRTModule:
-    """Convert an FX module to a TRT module
-    Args:
-        module: FX GraphModule to convert
-        inputs: Sequence of Tensors representing inputs to the module
-        settings: Compilation settings
-        name: TRT engine name
-    Returns:
-        _PythonTorchTensorRTModule or TorchTensorRTModule
-    """
-    # Specify module output data types to ensure TRT output types agree with
-    # that of the equivalent Torch module
+) -> TRTInterpreterResult:
     torch_inputs = get_torch_inputs(inputs, settings.device)
+    module.to(to_torch_device(settings.device))
```

Comment:
Is this line necessary? If it is only required for `convert_method_to_trt_engine`, then could it instead be called in `py/torch_tensorrt/_compile.py`?

Reply:
It's necessary because module and inputs should stay in the same device. I think we can let this line stay here because `convert_module` also needs to call it, and the `device` is read from `settings`. Same reason for `torch_inputs = get_torch_inputs(inputs, settings.device)`